marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/MintTokenEvent.hs

```diff
@@ -463,21 +473,34 @@ instance
       (MintTokenEventsMatchingQuery MintTokenBlockEvents)
       Core.ListIndexer
   where
-  query p (MintTokenEventsMatchingQuery predicate) ix = do
-    let convertError
-          :: Core.QueryError (Core.EventsMatchingQuery MintTokenBlockEvents)
-          -> Core.QueryError (MintTokenEventsMatchingQuery MintTokenBlockEvents)
-        convertError = \case
-          Core.NotStoredAnymore -> Core.NotStoredAnymore
-          (Core.IndexerQueryError t) -> Core.IndexerQueryError t
-          (Core.AheadOfLastSync r) -> Core.AheadOfLastSync r
-          (Core.SlotNoBoundsInvalid r) -> Core.SlotNoBoundsInvalid r
-
-    timedEventsE <- runExceptT $ Core.query p (Core.EventsMatchingQuery predicate) ix
-    timedEvents <- either (throwError . convertError) pure timedEventsE
-    pure $ sortEventsByOrderOfBlockchainAppearance timedEvents
-
-type instance Core.Result (QueryByAssetId event) = [Core.Timed (Core.Point event) event]
+  query p (MintTokenEventsMatchingQuery predicate) ix =
+    Core.withStability (view Core.point) ix $ do
+      let convertError
+            :: Core.ListIndexer MintTokenBlockEvents
+            -> Core.QueryError (Core.EventsMatchingQuery MintTokenBlockEvents)
+            -> m (Core.QueryError (MintTokenEventsMatchingQuery MintTokenBlockEvents))
+          convertError indexer = \case
+            Core.NotStoredAnymore -> pure Core.NotStoredAnymore
+            (Core.IndexerQueryError t) -> pure $ Core.IndexerQueryError t
+            (Core.AheadOfLastSync r) -> do
+              res <- case r of
+                Just val -> do
+                  r' <- Core.withStability (view Core.point) indexer (pure val)
+                  pure (Just r')
+                Nothing -> pure Nothing
+              pure $ Core.AheadOfLastSync res
+            (Core.SlotNoBoundsInvalid r) -> pure $ Core.SlotNoBoundsInvalid r
```

Comment:
isn't it the same as the conversion that we have below if so, can we just call it instead of duplicating the code?

Reply:
Since changing `withStability`'s input args, I don't think abstracting the logic (as it looks now, it definitely needed de-duping when you wrote the comment) is worthwhile.